File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.go
func longestConsecutive(nums []int) int {
    if len(nums)==0{
        return 0
    }
    hashmap:=make(map[int]bool)
    longest:=0
    for _,x :=range nums{
        hashmap[x]=true
    }
    for x:=range hashmap{
        if !hashmap[x-1]{
            cur:=x
            count:=1
            for hashmap[cur+1]{
                count++
                cur++
            }
            longest=max(longest,count)
        }
    }
    return longest
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashmap=defaultdict()
        longest=0
        for i in nums:
            if i in hashmap:
                hashmap[i]+=1
            else:
                hashmap[i]=1
        for key in hashmap:
            if key-1 not in hashmap:
                count=1
                x=key
                while x+1 in hashmap:
                    count+=1
                    x+=1
                longest=max(count,longest)
        return longest
